File: app/src/main/java/com/example/whattodo/model/CreditCard.java
package com.example.whattodo.model;

import javax.crypto.SecretKey;

public class CreditCard {

    private String numTargeta, expiritDate, nameOwner, CVV;
    private SecretKey key;
    private boolean isSelected;

    public CreditCard(){}

    public CreditCard(String numTargeta, String expiritDate, String nameOwner, String CVV, SecretKey key, boolean isSelected) {
        this.numTargeta = numTargeta;
        this.expiritDate = expiritDate;
        this.nameOwner = nameOwner;
        this.CVV = CVV;
        this.key = key;
        this.isSelected = isSelected;
    }

    public String getNumTargeta() {
        return numTargeta;
    }

    public String getExpiritDate() {
        return expiritDate;
    }

    public String getNameOwner() {
        return nameOwner;
    }

    public String getCVV() {
        return CVV;
    }

    public SecretKey getKey(){
        return key;
    }

    public boolean isSelected(){
        return isSelected;
    }

    public void setSelected(boolean isSelected){
        this.isSelected = isSelected;
    }

}

File: app/src/main/java/com/example/whattodo/model/CreditCard.kt
package com.example.whattodo.model

import javax.crypto.SecretKey

class CreditCard {
    var numTargeta: String? = null
        private set
    var expiritDate: String? = null
        private set
    var nameOwner: String? = null
        private set
    var cVV: String? = null
        private set
    var key: SecretKey? = null
        private set
    var isSelected = false

    constructor() {}
    constructor(numTargeta: String?, expiritDate: String?, nameOwner: String?, CVV: String?, key: SecretKey?, isSelected: Boolean) {
        this.numTargeta = numTargeta
        this.expiritDate = expiritDate
        this.nameOwner = nameOwner
        this.cVV = CVV
        this.key = key
        this.isSelected = isSelected
    }
}